jdaviz/configs/imviz/helper.py

```diff
@@ -653,34 +656,30 @@ def link_image_data(app, link_type='pixels', wcs_fallback_scheme=None, wcs_use_a
 
     app._link_type = link_type
     app._wcs_use_affine = wcs_use_affine
-    viewer_ref = app._jdaviz_helper.default_viewer.reference
-    viewer_item = app._get_viewer_item(viewer_ref)
-
-    viewer_item['reference_data_label'] = refdata.label
-
-    if link_plugin is not None:
-        # Only broadcast after success.
-        app.hub.broadcast(LinkUpdatedMessage(link_type,
-                                             wcs_fallback_scheme == 'pixels',
-                                             wcs_use_affine,
-                                             sender=app))
-
-        if insert_base_wcs_layer:
-            # update all viewer items with reference data:
-            for viewer_id in app.get_viewer_ids():
-                viewer_item = app._get_viewer_item(viewer_id)
-                viewer_item['reference_data_label'] = refdata.label
-
-        # reset the progress spinner
-        link_plugin.linking_in_progress = False
 
     for viewer in app._viewer_store.values():
         wcs_linked = link_type == 'wcs'
         # viewer-state needs to know link type for reset_limits behavior
         viewer.state.linked_by_wcs = wcs_linked
         # also need to store a copy in the viewer item for the data dropdown to access
+        viewer_item = app._get_viewer_item(viewer.reference)
+
+        viewer_item['reference_data_label'] = refdata.label
         viewer_item['linked_by_wcs'] = wcs_linked
 
+        if insert_base_wcs_layer:
+            viewer_item['reference_data_label'] = refdata.label
+
```

Comment:
Aren't the first and last lines of this section redundant?

Reply:
hmmm yes.  I think we always want to do this, right?  Can I just remove the second instance?

Reply:
I think so ✅

Reply:
done